### PR TITLE
[mapreduce-tidb-connector] mapreduce read from tikv

### PR DESCRIPTION
**Pull request by mengxin9014**
support TiDBInputFormat for  mapreduce to read from Tikv

### CODE REVIEW

File: check_style.xml [hunk 1]
```diff
@@ -228,7 +228,7 @@
         value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF,
                     PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF"/>
     </module>
-    <module name="OverloadMethodsDeclarationOrder"/>
+<!--    <module name="OverloadMethodsDeclarationOrder"/>-->
```
Comment (marsishandsome) [inline check_style.xml:231]: revert this change?

File: mapreduce/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ -p   ${PASSWORD}  \
+ -dn  ${TIDB_DATABASE} \
+ -n   ${TABLE_NAME} \
+ -f   ${FIELD1} \
+ -f   ${FIELD2} \
+ -t   ${TABLE_NAME}
+ -ts  ${SNAPSHOT_TIMESTAMP} \
+ -l   ${LIMIT} 
+```
+###example
```
Comment (marsishandsome) [inline mapreduce/README.md:32]:
```suggestion
### example
```

File: mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputFormat.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    this.tableHandleInternal = new TableHandleInternal(
+        UUID.randomUUID().toString(), databaseName, tableName);
+    List<ColumnHandleInternal> columns = clientSession.getTableColumns(tableHandleInternal)
+        .orElseThrow(() -> new NullPointerException("columnHandleInternals is null"));
+    Map<String, Integer> nameAndIndex = new HashMap<>();
+
+    IntStream.range(0, columns.size())
+        .forEach(i -> nameAndIndex.put(columns.get(i).getName(), i));
+
+    String[] fieldNames = getLowCaseFields(dbConf.getInputFieldNames());
```
Comment (sunxiaoguang) [inline mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputFormat.java:105]: Arrays.stream(dbConf.getInputFieldNames()).map(String::toLowerCase).toArray(String[]::new)?

File: mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputSplit.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  @Override
+  public long getLength() {
+    return 0;
+  }
+
+  @Override
+  public String[] getLocations() {
+    return new String[0];
```
Comment (sunxiaoguang) [inline mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputSplit.java:56]: How about create a final static empty array and reuse it since it's immutable and can be shared.

File: check_style.xml (hunk 1 above)
Comment (mengxin9014) [reply check_style.xml]: done

File: mapreduce/README.md (hunk 2 above)
Comment (mengxin9014) [reply mapreduce/README.md]: done

File: mapreduce/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
+# MapReduce-TiDB-Connector
+
+## Build
+
+```bash
+git clone git@github.com:pingcap-incubator/TiBigData.git
+cd TiBigData
+mvn clean package -DskipTests -am -pl mapreduce/mapreduce-base
+cp mapreduce/mapreduce-2.0/target/mapreduce-tidb-connector-base-0.0.3-SNAPSHOT.jar ${HOME}/lib
```
[the remark this answers is not in the document]
Comment (mengxin9014) [reply mapreduce/README.md]: done

File: mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/MapreduceCmd.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import io.tidb.bigdata.tidb.ClientConfig;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+
+public class MapreduceCmd {
+  @Parameter(names = {"-filed", "-f"}, description = "fileds need query")
```
[the remark this answers is not in the document]
Comment (mengxin9014) [reply mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/MapreduceCmd.java]: done

File: mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputFormat.java (hunk 3 above)
Comment (mengxin9014) [reply mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputFormat.java]: done

File: mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputSplit.java (hunk 4 above)
Comment (mengxin9014) [reply mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBInputSplit.java]: done
